pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -312,12 +368,14 @@ type SourceInfo struct {
 //
 //	Artifact (GeneratedBy) -> Step (Input) -> Source
 func getSourceInfo(sourceID strfmt.UUID, lookup map[strfmt.UUID]interface{}) (SourceInfo, error) {
-	source, ok := lookup[sourceID].(*model.Source)
+	node := lookup[sourceID]
```

Comment:
If you add another `ok` check here we could skip the following logic if the `sourceID` is not a key in the `lookup` map.